robot/client/client.go

```diff
@@ -161,7 +152,7 @@ func (rc *RobotClient) handleUnaryDisconnect(
 	err := invoker(ctx, method, req, reply, cc, opts...)
 	// we might lose connection before our background check detects it - in this case we
 	// should still surface a helpful error message.
-	if isClosedPipeError(err) {
```

Comment:
These "if closed pipe" conditionals were a little abstract before. Can we tighten up the documentation here? I'd love if we could list out exactly which errors bubble up here (obvious `ErrDisconnected` is one). And how this conditional changes the behavior in a desirable way.

Reply:
Spoke a bit with @dgottlieb offline. I did some sleuthing into what errors can bubble up to this point; I do not think we're catching them all. Left a `TODO` with a reference to https://viam.atlassian.net/browse/RSDK-9333 and a `NOTE` explaining my findings/thoughts.

For now, I think we should capture _both_ `rpc.ErrDisconnected` and `io.ErrClosedPipe` as possible disconnect manifestations, so we are not _less_ sensitive than before. Per RSDK-9333, we should likely capture more than that and transform the errors into something more informative than an `Unavailable` gRPC error.

The test that was failing in a previous CI run was indeed "injecting" an `io.ErrClosedPipe` and expecting us to transform it into a gRPC error with code `Unavailable`. Left that test as-is.